src/main/scala/mimir/adaptive/AdaptiveSchemaManager.scala

```diff
@@ -27,56 +27,62 @@ class AdaptiveSchemaManager(db: Database)
     }
   }
 
-  def create(schema: String, mlensType: String, query: Operator, args: Seq[Expression]) = 
+  def create(schema: String, mlensType: String, query: Operator, args: Seq[Expression]) =
   {
     val constructor = MultilensRegistry.multilenses(mlensType)
     val config = MultilensConfig(schema, query, args);
-    val models = constructor.initSchema(db, config);
-    
-    db.backend.update(s"""
-      INSERT INTO $dataTable(NAME, MLENS, QUERY, ARGS) VALUES (?,?,?,?)
-    """, Seq(
-      StringPrimitive(schema),
-      StringPrimitive(mlensType),
-      StringPrimitive(Json.ofOperator(query).toString),
-      StringPrimitive(Json.ofExpressionList(args).toString)
-    ))
-
-    // Persist the associated models
-    for(model <- models){
-      if(model.isInstanceOf[mimir.models.NeedsReconnectToDatabase])
-        model.asInstanceOf[mimir.models.NeedsReconnectToDatabase].reconnectToDatabase(db)
-      db.models.persist(model, s"MULTILENS:$schema")
+    if(mlensType == "DETECT_HEADER"){
+      val models1 = constructor.checkheader(db, config);
+    }
+    else{
+      var models = constructor.initSchema(db, config);
+
+
+      db.backend.update(s"""
+        INSERT INTO $dataTable(NAME, MLENS, QUERY, ARGS) VALUES (?,?,?,?)
+      """, Seq(
+        StringPrimitive(schema),
+        StringPrimitive(mlensType),
+        StringPrimitive(Json.ofOperator(query).toString),
+        StringPrimitive(Json.ofExpressionList(args).toString)
+      ))
+
+      // Persist the associated models
+      for(model <- models){
+        if(model.isInstanceOf[mimir.models.NeedsReconnectToDatabase])
+          model.asInstanceOf[mimir.models.NeedsReconnectToDatabase].reconnectToDatabase(db)
+        db.models.persist(model, s"MULTILENS:$schema")
+      }
```

Comment:
You should not need to modify this function to introduce a new adaptive schema.  All code specific to DETECT_HEADER new should be isolated in the constructor.